pkg/kubelet/images/image_manager.go

```diff
@@ -136,6 +138,19 @@ func (m *imageManager) EnsureImageExists(pod *v1.Pod, container *v1.Container, p
 			msg := fmt.Sprintf("image pull failed for %s because the registry is unavailable.", container.Image)
 			return "", msg, imagePullResult.err
 		}
+		// Pull image failure because image is corrupt: https://github.com/kubernetes/kubernetes/issues/46184
+		if strings.Contains(imagePullResult.err.Error(), "rename") && strings.Contains(imagePullResult.err.Error(), "directory not empty") {
```

Comment:
Hi @smarterclayton 

It's my understanding that the Godeps.json references that github.com/docker/engine-api is being used to manage Docker images, etc. This project has been deprecated and hasn't been contributed to since 2016.

Updating the CI environment to use Docker 1.13.X should also address this issue. Not sure if the engine-api in this deprecated code base will support 1.13.X.